### PR TITLE
Update using.html

### PR DESCRIPTION
**Pull request by eah13**
A more polished draft of the Using Skulpt page.  As discussed in #454 it's split into three sections of increasing complexity.

The elements I'd love comments on that will be used on other pages if they're generally seen as successful:
- the floating nav bar that appears when you hit the first section
- the callout panel at the top

This is by no means a final draft. Merging this PR will not deploy to production, just take another step towards the final layout/content. I've not done much editing inside the three sections and expect to give those more love once the site in general is laid out.  So, in other words, layout and organization comments are more helpful than wordsmithing at this particular juncture. :)

**Update: I've temporarily deployed this PR to http://elliotthauser.com/skulpt/ for ease of review**

---

Since this is the first PR for the website, here are some quick instructions on building the site. ~~Let me know if anything's amiss- this is from memory.~~ Update: added gem and jekyll install instructions:

Make sure you have rubygems installed (`sudo apt-get install rubygems-integration` on Ubuntu 14 - or follow instructions [here](https://rubygems.org/pages/download)).  Then:

```
gem install github-pages
```

This is a meta-bundle that gets you everything you need for your setup to work like github's prod including jekyll and its various plugins.

Next, if you don't have a copy of the gh-pages branch you'll need to create an orphan branch to pull this code to:

```
git checkout --orphan pages
git rm -rf .    
git pull git@github.com:trinketapp/skulpt.git pages
```

If you already have gh-pages you can `git checkout -b pages` from that branch. 

With the code safely on your system you're now ready to

```
npm install
bundle install
bundle exec jekyll serve --watch
```

If these commands doesn't work for you, [see the pre-reqs](https://github.com/skulpt/skulpt/tree/gh-pages#skulpt-project-homepage)

This will build the site in the gitignored `_site` directory.  This jekyll build, by the way, is essentially what Github does when we commit a change to the gh-pages branch. That's why we don't need to include the built static site in the repo.

If you'll be making SCSS changes you'll want to open a new terminal tab and 

```
grunt
```

More on hacking in the [README](https://github.com/skulpt/skulpt/tree/gh-pages#skulpt-project-homepage)

~~Side note: If I committed these to trinket's `gh-pages` branch they'd be live at trinketapp.github.io/skulpt.  I've not done that to avoid a mess if the PR isn't accepted.  I could start committing changes there to have an easily reviewable live page if people like that more.~~ I'll deploy future PRs to http://elliotthauser.com/skulpt/ via the eah13 fork, while doing dev work on the trinketapp fork to avoid any conflicts in the code.

### CODE REVIEW

**Comment by waywaaard** [comment]
I would like to see a link on the gh-pages branch to easily reach the draft: https://skulpt.github.io/skulpt/

**Comment by bnmnetp** [comment]
I followed the instructions, but I'm not getting the updated using page.

**Comment by bnmnetp** [comment]
Oh, nevermind, the new stuff is in a PR.

**Comment by eah13** [comment]
Gross- just figured I did a stupid thing with an extraneous commit (a503887) that reverted all my changes.  I removed the offending commit on this PR and you can `git reset --hard 301dc07e` locally.

Anyways, I figure i'll use my personal fork's gh-pages branch as a dumping ground for PRs so people can preview them live.  Here that is:

http://elliotthauser.com/skulpt/using.html

(it's showing on elliotthauser.com because of the ANAME I use for my own github pages User site)

Here's the fancy menubar you should now see:

screenshot 2015-06-02 at 8 48 58 am

Sorry but we're good now!  I can deploy one PR at a time to my fork and reviews should be much easier.

**Comment by waywaaard** [comment]
Maybe we should also out the gitter link on the homepage.

**Comment by eah13** [comment]
Good thought. And definitely on the contributing page. And/or perhaps there's a 'get in touch' page or section of a page that wants to happen and has links to bitter and Google groups.

**Comment by LeszekSwirski** [comment]
I like it a lot! A few comments:

* The modules section feels like it belongs in more general skulpt documentation
* The floating header makes the first section "jump" down when you scroll to it
* The Trinket paragraph has weird line breaks (or a weird width)
* I would rephrase "using Skulpt with HTML" to just "using Skulpt" (since this is the "raw" way of doing it), and the trinket section to something like "embedding a ready-made Skulpt editor".
* I'm not sure on the order of the above two sections either; on the one hand, I see that what you're doing is "easy" followed by "harder", but the other way around would be "default" followed by "external alternative".
* I love the callout table of contents
* The link colour is a bit subtle, I can barely tell when links are there.

**Comment by bnmnetp** [comment]
I agree with @LeszekSwirski this is a huge step forward, but some of the language should be discussed/tweaked.

* Embed Skulpt  Well this is what we do no matter what.  We are embedding skulpt in a web page, we are just talking about different methods of embedding.

Maybe:

* the Skulpt API  -- which demonstrates the raw/normal way
* Third Party Solutions  -- which highlights trinket

Now putting on my marketing hat for a moment, I'm thinking about messaging and what we want to convey with this new site.

1.  Skulpt is a solid and reliable method for learning Python in the browser
2.  Skulpt is a solid implementation of Python in the browser
3.  Skulpt is a vibrant and active open source project
4.  Skulpt needs and welcomes your help as a developer and gives you the resources to get up to speed quickly.

Important questions to answer:

How do I use skulpt on my own site?
How do I use skulpt as a teaching resource?
Who uses skulpt?
Why you should choose skulpt over other alternatives?

**Comment by eah13** [comment]
Thanks everyone for the great comments.  I haven't gotten my one hour a day target this week but will try to this coming week.

Quick summary of what I've taken from the feedback so far:

* using.html should be split into using the Skulpt API and Third-party solutions, with API info first and foremost.
* @bnmnetp's marketing hat produced some really great key takeaways and important questions to answer.  I'll see if I can incorporate the takeaways lightly into the homepage and pick one or more to emphsize on each of the main pages.
* Of the important questions to answer, I think How do I use Skulpt is answered on using.html.  WHo uses skulpt and how do I teach with it should be answered by gallery.html.  Why should you choose skulpt is an interesting question not addressed on the current pages.  I'll need to think about how it might be addressed- is it another link in the Nav?  A comparison grid of features on the homepage?

If I start addressing some of these on this branch then this PR may grow into touching much more than using.html.

**Comment by eah13** [comment]
Oh and another thing:

* thanks to @waywaaard for mentioning the Gitter link.  That may make sense on the homepage or on Contributing, depending on who we see clicking on it.

**Comment by waywaaard** [comment]
Maybe we should also list some of the papers published about skulpt or using it.

**Comment by eah13** [comment]
Great idea.  Know of any?

Fires keep coming up that are slowing down this work but it's still on the list.  Keep the ideas coming in the meantime.

**Comment by waywaaard** [comment]
For example: http://dl.acm.org/citation.cfm?id=2325335
or
http://dl.acm.org/citation.cfm?id=2538862.2538908&coll=DL&dl=GUIDE&CFID=669427394&CFTOKEN=37146055

**Comment by bnmnetp** [comment]
Here's another project to add to our list..  https://github.com/PETComputacaoUFPR/skulpt-console

**Comment by bnmnetp** [comment]
I would love to revisit the new home page idea sometime in the future.

Perhaps we should just get skulpt.io and redirect skulpt.org there.

**Comment by albertjan** [comment]
@bnmnetp lets reopen and merge this it's into the gh-pages branch that's not used and this is better then what's in there now.

We could point skulpt.org at a GitHub page right?

**Comment by rixner** [comment]
You can definitely point a custom domain at a GitHub page - I do it for another project.  I think that is the way to go here.  I don't think there's any good reason not to be using GitHub pages...

**Comment by bnmnetp** [comment]
Yes, I can definitely set up a redirect from skulpt.org to github pages.  But to have a custom domain point to your github page you need to have control of the domain.  I've asked Scott many times if he would be willing to transfer it, and he always says he'll just keep paying for it.

**Comment by bnmnetp** [comment]
OK You can see the state of the "new" homepage at:  http://skulpt.github.io/skulpt/

There are still things to clean up before I would say we ought to move over to that. Missing images in the gallery are primary, and we need to clean up stuff about external libraries. etc.


Lots of people use skulpt.org to  as a server for skulpt*.js   We've never encouraged that, and it was out of date for a few years, so I guess its not really something to worry too much about.

**Comment by rixner** [comment]
@bnmnetp maybe Scott would be willing to update the skulpt.org DNS entries to point to github pages, even if he wants to still retain control of the domain name?
